x-pack/plugins/elastic_assistant/server/routes/attack_discovery/post_attack_discovery.ts

```diff
@@ -91,66 +110,74 @@ export const postAttackDiscoveryRoute = (
             latestReplacements = { ...latestReplacements, ...newReplacements };
           };
 
-          // get the attack discovery tool:
-          const assistantTools = (await context.elasticAssistant).getRegisteredTools(pluginName);
-          const assistantTool = assistantTools.find((tool) => tool.id === 'attack-discovery');
+          const assistantTool = getAssistantTool(
+            (await context.elasticAssistant).getRegisteredTools,
+            pluginName
+          );
+
           if (!assistantTool) {
             return response.notFound(); // attack discovery tool not found
           }
 
-          const traceOptions = {
-            projectName: langSmithProject,
-            tracers: [
-              ...getLangSmithTracer({
-                apiKey: langSmithApiKey,
-                projectName: langSmithProject,
-                logger,
-              }),
-            ],
-          };
-
-          const llm = new ActionsClientLlm({
-            actions,
-            connectorId,
-            llmType: getLlmType(actionTypeId),
-            logger,
-            request,
-            temperature: 0, // zero temperature for attack discovery, because we want structured JSON output
-            timeout: CONNECTOR_TIMEOUT,
-            traceOptions,
-          });
-
           const assistantToolParams = getAssistantToolParams({
+            actions,
             alertsIndexPattern,
             anonymizationFields,
+            apiConfig,
             esClient,
             latestReplacements,
+            connectorTimeout: CONNECTOR_TIMEOUT,
             langChainTimeout: LANG_CHAIN_TIMEOUT,
-            llm,
+            langSmithProject,
+            langSmithApiKey,
+            logger,
             onNewReplacements,
             request,
             size,
           });
 
           // invoke the attack discovery tool:
           const toolInstance = assistantTool.getTool(assistantToolParams);
-          const rawAttackDiscoveries = await toolInstance?.invoke('');
-          if (rawAttackDiscoveries == null) {
-            return response.customError({
-              body: { message: 'tool returned no attack discoveries' },
-              statusCode: 500,
-            });
-          }
 
-          const { alertsContextCount, attackDiscoveries } = JSON.parse(rawAttackDiscoveries);
+          const { currentAd, attackDiscoveryId } = await updateAttackDiscoveryStatusToRunning(
+            dataClient,
+            authenticatedUser,
+            apiConfig,
+            latestReplacements
+          );
+
+          toolInstance
+            ?.invoke('')
+            .then(async (rawAttackDiscoveries) =>
```

Comment:
Consider not mixing `await` and `.then` styles in the same function

Reply:
https://github.com/elastic/kibana/pull/184949/commits/803d19adc32878808d32ec733846e2199dc1f236